File: 20-02-2024/jogos-educativos/salario.c
#include <stdio.h>

float gettotal(float price, int sold) {
    return price * sold;
}

float getwage(float total, float bonus) {
    return (total / 2) + bonus;
}

float calcbonus(float actual, float agg, int times, int index) {
    if (times == index) {
        return agg;
    }

    return calcbonus(actual * 0.92, agg + (actual * 0.08), times, (index + 1));
}

float getbonus(float total, int sold) {
    int times = sold / 15;

    if (times == 1) {
        return total * 0.08;
    }

    return calcbonus(total, 0, times, 0);
}

int main() {
    float price = 19.9;
    int sold;

    printf("Informe o número de vendas: ");
    scanf("%d", &sold);

    float total = gettotal(price, sold);
    float bonus = getbonus(total, sold);
    float wage = getwage(total, bonus);

    printf("%.2f\n", total);
    printf("%.2f\n", bonus);
    printf("%.2f\n", wage);

    return 0;
}

File: 20-02-2024/jogos-educativos/Salario.java
import java.util.Scanner;

public class Salario {

    public static void main(String[] args) {
        final double price = 19.9;

        try (Scanner scanner = new Scanner(System.in)) {
            System.out.println("Informe o nº de vendas:");

            final int sold = scanner.nextInt();

            final double total = price * sold;
            final double bonus = getBonus(total, ((int) sold / 15));
            final double wage = (total / 2) + bonus;

            System.out.println(String.format("%.2f", total));
            System.out.println(String.format("%.2f", bonus));
            System.out.println(String.format("%.2f", wage));
        }
    }

    public static double getBonus(double total, int times) {
        if (times == 0) {
            return total * 0.08;
        }

        double agg = 0;
        double actual = total;

        for (int i = 0; i < times; i++) {
            agg += actual * 0.08;

            actual *= 0.92;
        }

        return agg;
    }

}